libcontainer/nsenter/nsexec.c

```diff
@@ -966,7 +632,7 @@ void nsexec(void)
 				sane_kill(stage2_pid, SIGKILL);
 				bail("failed to sync with parent: write(SYNC_RECVPID_PLS)");
 			}
-			if (write(syncfd, &stage2_pid, sizeof(stage2_pid)) != sizeof(stage2_pid)) {
+			if (dprintf(syncfd, "%d\n", stage2_pid) < 0) {
```

Comment:
This might not work in all cases, for example because `dprintf` may do 2 write(2) calls instead of one. Which may or may not be ok.